libsql-server/src/replication/primary/logger.rs

```diff
@@ -84,6 +85,21 @@ unsafe impl WalHook for ReplicationLoggerHook {
         tracing::trace!("Last valid frame before applying: {last_valid_frame}");
         let ctx = Self::wal_extract_ctx(wal);
 
+        let mut binding = ctx.bottomless_replicator.clone();
+        let replicator_mutex: Option<&mut Replicator>;
+        let mut replicator_lock: MutexGuard<Option<Replicator>>;
+        if let Some(replicator) = binding.as_mut() {
+            replicator_lock = replicator.lock().unwrap();
+            if replicator_lock.is_none() {
+                tracing::error!("fatal error: expected a replicator, exiting");
+                std::process::abort()
```

Comment:
do not abort the process here, just return an io error instead

Reply:
we just need to prevent the write, no need to shutdown the whole instance

Reply:
Thats a big oops from my side. I had done this at four places and replaced all of them to return `SQLITE_IOERR`